src/sos_utils.py

```diff
@@ -0,0 +1,77 @@
+"""Collections of sos related utilities."""
+import glob
+import logging
+from pathlib import Path
+from subprocess import DEVNULL, CalledProcessError, check_call
+from typing import List, Optional, Union
+
+from uploader import Uploader
+
+logger = logging.getLogger(__name__)
+
+
+class ReportNotFoundError(Exception):
+    """Report not found exception."""
+
+
+class SoSCollectHelper:
```

Comment:
**praise:** 

I like what we implement here. We hide sos behind this Helper class, so it's much easy to write test.